File: frontend/src/lib/column/model/HitRate.ts
import SingleTableData from "src/component/stats/internal/table_data/SingleTableData.svelte";
import { AbstractStatsColumn } from "src/lib/column/intetface/AbstractStatsColumn";
import type { data } from "wailsjs/go/models";

export class HitRate extends AbstractStatsColumn<string> {
  constructor() {
    super("hit_rate", "ship");
  }

  displayValue(player: data.Player): string {
    const hitRate = this.playerStats(player).ship.hit_rate;

    const main = hitRate.main_battery.format(this.digit());
    const torps = hitRate.torpedoes.format(this.digit());

    return `${main}|${torps}`;
  }

  getTableDataComponent() {
    return SingleTableData;
  }

  getCssClass(): string | undefined {
    return "text-center";
  }
}

File: backend/domain/model/vo/hit_rate.go
package vo

type HitRate struct {
	ValueObject2[Percentage, Percentage]
}

func NewHitRate(mainBattery Percentage, torps Percentage) HitRate {
	vo2 := ValueObject2[Percentage, Percentage]{mainBattery, torps}
	return HitRate{vo2}
}
